src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ActiveConfiguredProjectsLoader.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright(c) Microsoft.All Rights Reserved.Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    ///     Force loads the active <see cref="ConfiguredProject"/> objects so that any configured project-level 
+    ///     services, such as evaluation and build services, are started.
+    /// </summary>
+    internal class ActiveConfiguredProjectsLoader : OnceInitializedOnceDisposed
+    {
+        private readonly UnconfiguredProject _project;
+        private readonly IActiveConfigurationGroupService _activeConfigurationGroupService;
+        private ActionBlock<IProjectVersionedValue<IConfigurationGroup<ProjectConfiguration>>> _targetBlock;
+        private IDisposable _subscription;
+
+        [ImportingConstructor]
+        public ActiveConfiguredProjectsLoader(UnconfiguredProject project, IActiveConfigurationGroupService activeConfigurationGroupService)
+            : base(synchronousDisposal:true)
+        {
+            _project = project;
+            _activeConfigurationGroupService = activeConfigurationGroupService;
+            _targetBlock = new ActionBlock<IProjectVersionedValue<IConfigurationGroup<ProjectConfiguration>>>(OnActiveConfigurationsChanged);
+        }
+
+        [ProjectAutoLoad(ProjectLoadCheckpoint.ProjectInitialCapabilitiesEstablished)]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharpLanguageService)]
+        public Task InitializeAsync()
+        {
+            Initialize();
```

Comment:
Should probably use `EnsureInitialized()`

Reply:
Nice catch.